cylc/flow/data_store_mgr.py

```diff
@@ -2525,29 +2521,53 @@ def delta_task_ext_trigger(
         ext_trigger.time = update_time
         self.updates_pending = True
 
-    def delta_task_xtrigger(self, sig, satisfied):
-        """Create delta for change in task proxy xtrigger.
+    def _delta_xtrigger(
```

Comment:
This will conflict with #6671 ?

Reply:
We've got a lot of xtrigger stuff going on ATM, I would like to get 6671 reviewed and in soon (especially as it unblocks working on xtriggers in the GUI which is being raised as a high requirement)

Reply:
The trouble is, the existing method (which you've highlighted here) is for recording satisfied xtriggers in the DB, regardless of which tasks depend on them. Whereas the new method I've added here is to record a specific task's dependence on an xtrigger - so the old name is more appropriate for the new method. I can revert this and come up with a new name, supported by explanatory docstrings, if this change causes more trouble than it's worth?

Reply:
Ok, I have now re-implemented my xtrigger branch (https://github.com/cylc/cylc-flow/pull/6671) in a way that is compatible with your changes here.

It just needs your new `delta_task_xtrigger` bit jammed in around here:

https://github.com/cylc/cylc-flow/pull/6671/files#diff-aaaccca49464232ef4889b5092201eb44575e83602d68b38f9608755ae90982aR2561

Reply:
>We have both attempted to solve a similar problem for different reasons, months apart. But, unfortunately, it is gonna conflict bad!

Actually only about 1 week apart according to the dates on our PRs. In that week, I didn't look closely enough at your PR, sorry!, because the headline suggests it's UI focused, whereas this one is (I thought) just about task dependence on xtriggers, in the scheduler. But yes, I can see the overlap now.

> This has actually conflicted with yet another xtrigger PR - https://github.com/cylc/cylc-flow/pull/6721 - which makes acquiring the itask difficult.

That one was only a one-line bug fix though.